File: ruby/lexer.rb

module Lexer

	TOKENS = [
		'title:',
		'subtitle:',
		'for:',
		'ingredients:',
		'procedure:'
	]

	class Lexer

		@current_field = "unfielded"
		@fields = {}

		def initialize(filename = "")

			@fields = Hash.new
			if filename == ""
				raise "No filename for lexer!"
			end
			
			file = File.open("../tests/eggs.rcp")
			file_data = file.read
			file.close

			for line in file_data.split "\n" do
				lex_line line
			end
		end

		def lex_line(line)
			line_split = line.split " "

			if TOKENS.include? line_split[0]
				@current_field = line_split[0]
				line_split = line_split.drop 1
			end
			
			
			if @fields.key @current_field == nil
				@fields[@current_field] = Array.new
			end

			@fields[@current_field].push line_split

		end


	end
end

File: py/lexer.py
from locale import currency


TOKEN = [
	"title:",
	"subtitle:",
	"for:",
	"ingredients:",
	"procedure:"
]

class Lexer:

	fields = {}
	current_field = "unfielded"

	def __init__(self, file) -> None:
		with open(file, "r") as handle:
			file_data = handle.readlines()
		
		for line in file_data:
			self.LexLine(line)



	def LexLine(self, line: str):
		
		line = line.strip() 
		line_split = line.split(" ")

		if (line_split[0] in TOKEN):
			self.current_field = line_split[0]
			line_split = line_split[1:]

		if (self.current_field not in self.fields):
			self.fields[self.current_field] = []


		self.fields[self.current_field].append(" ".join(line_split))
